### PR TITLE
os/bluestore: simplify LRUCache::trim()

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

File: src/include/intarith.h [hunk 1]
```diff
@@ -16,23 +16,35 @@
...
+/*
+ * Return "n" rounded down to the "d" boundary.
+ * Note that "d" must be a power of 2.
+ * eg, ROUND_DOWN_TO(1200, 1024) == 1024 (1*d)
+ * eg, ROUND_DOWN_TO(1024, 1024) == 1024 (1*d)
+ * eg, ROUND_DOWN_TO(0x1234, 0x100) == 0x1200 (0x12*d)
+ * eg, ROUND_DOWN_TO(0x5600, 0x100) == 0x5600 (0x56*d)
+ */
+#ifndef ROUND_DOWN_TO
+#define ROUND_DOWN_TO(n, d) ((n) & -(d))
```
Comment (liewegas) [inline src/include/intarith.h]: ROUND_DOWN_TO_POW2 ?

File: src/os/bluestore/BlueStore.cc [hunk 2]
```diff
@@ -1749,13 +1749,28 @@ int BlueStore::_open_fm(bool create)
       uint64_t start = ROUND_UP_TO(reserved, min_alloc_size);
       uint64_t max_b = g_conf->bluestore_debug_prefragment_max / min_alloc_size;
       float r = g_conf->bluestore_debug_prefill;
+      r /= 1.0 - r;
+
       while (start < end) {
+
 	uint64_t l = (rand() % max_b + 1) * min_alloc_size;
-	if (start + l > end)
+        l = ROUND_UP_TO(l, min_alloc_size);
```
Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: l = something * min_alloc_size .. shouldn't need to round up?

**Comment by liewegas** [comment]
second and third commits look good!

File: src/include/intarith.h [hunk 3]
```diff
@@ -16,23 +16,35 @@
...
 #endif
 
 #ifndef DIV_ROUND_UP
-# define DIV_ROUND_UP(n, d)  (((n) + (d) - 1) / (d))
+#define DIV_ROUND_UP(n, d)  (((n) + (d) - 1) / (d))
 #endif
 
 #ifndef ROUND_UP_TO
-# define ROUND_UP_TO(n, d) ((n)%(d) ? ((n)+(d)-(n)%(d)) : (n))
+#define ROUND_UP_TO(n, d) ((n)%(d) ? ((n)+(d)-(n)%(d)) : (n))
```
Comment (xiexingguo) [inline src/include/intarith.h]: Exactly , but  `DIV_ROUND_UP`(at line 27) already did this.

Others are all fixed.

**Comment by xiexingguo** [comment]
change set:

- introduce stop to avoid potential dead loop during debug prefill.

**Comment by xiexingguo** [comment]
@liewegas  Move the first change into https://github.com/ceph/ceph/pull/10128.

**Pull request merged**